core/tests/compile-fail/component_derive_tests.rs

```diff
@@ -0,0 +1,21 @@
+use kompact::prelude::*;
+
+#[derive(Debug)]
+struct TestPort;
+impl Port for TestPort {
+    type Indication = String;
+    type Request = usize;
+}
+
+#[derive(ComponentDefinition)] //~ ERROR: Ambiguous port type: There are multiple fields with type TestPort (["provided_test_port1", "provided_test_port2"]). You cannot derive ComponentDefinition in these cases, as you must resolve the ambiguity manually.
+struct DuplicatePortComponent {
+    ctx: ComponentContext<Self>,
+    provided_test_port1: ProvidedPort<TestPort>,
+    provided_test_port2: ProvidedPort<TestPort>,
+    required_test_port1: RequiredPort<TestPort>,
+    required_test_port2: RequiredPort<TestPort>,
+}
```

Comment:
Wait, but looking at your output, it seems to be working correctly, right? It's expecting an error message about ambiguous ports, but since you are using different generics you are not getting that error message. (Just all the others about the stuff I dind't bother implementing^^).

Reply:
Sorry for late response, for me it's good, I am happy with the output 👍